debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/junit/MongoDbDatabaseProvider.java

```diff
@@ -78,6 +103,17 @@ public static MongoDbDeployment externalOrDockerReplicaSet() {
         return platform.provider.get();
     }
 
+    /**
+     * Creates MongoDB replica-set abstraction either for external database or a local MongoDB replica-set container with the provided container Network.
+     *
+     * @return MongoDb replica-set deployment
+     */
+    public static MongoDbDeployment externalOrDockerReplicaSet(Network network) {
+        NETWORK = network;
```

Comment:
This just doesn't seem like something we should do.

Reply:
The reason for this is that all those methods are already static. That's just the sugar on top of the cake, no?

Reply:
That switch is not magical -- is purely based on property and can be backtraced in IDE easily. 

Additionally there is a significant difference between static method and static attribute -- just one of those creates a state.

Reply:
Not really. A static method, which in this case has no option to pass custom config changes / a custom builder or parameters is a shortcoming. I think it would be better to move out some instance config/creation code from `MongoDbDatabaseProvider` into the `MongoDbReplicaSet` and `MongoDbShardedCluster` classes and refactor `MongoDbPlatform` to allow passing a custom builder that reconfigures the default config when provided. That way that static intermediate state is not bad.

On the other hand: It's not too bad to have this static state for the Network in `MongoDbDatabaseProvider`. This class lives inside the `test` context only and I assume there's only one network per test run.

One suggestion:
```
    public static MongoDbReplicaSet dockerReplicaSet() {
        var replicaSetBuilder = dockerReplicaSetBuilder();
        if (NETWORK != null) {
            replicaSetBuilder.network(NETWORK);
        }
        var replicaSet = replicaSetBuilder.build();
        NETWORK = null;
        return replicaSet;
    }
```
Would that solve the side effects you mentioned?

Reply:
@jcechace wrote above:


> E.g. where is the `NETWORK` attribute set back to null? How do you ensure that the tests which will run after / intertwined with yous will not suddenly stat using that network?

see here. I added reset.

Reply:
So you have a test class which sets the NETWORK, then you have a different test class which wants  a fresh new network / the default and so it calls `dockerReplicaSet()`. What happens? It gets the old one. Or am I reading it wrong?

BTW: this doesn't really solve anything, it just digs the pit of temporary workarounds even deeper.